### PR TITLE
Fixed #24468 -- Made signed cookies cache backend resilient to unpickling exceptions.

### PR DESCRIPTION
**Pull request by timgraham**

### CODE REVIEW

File: django/contrib/sessions/backends/signed_cookies.py [hunk 1]
```diff
@@ -17,7 +17,9 @@ def load(self):
                 # This doesn't handle non-default expiry dates, see #19201
                 max_age=settings.SESSION_COOKIE_AGE,
                 salt='django.contrib.sessions.backends.signed_cookies')
-        except (signing.BadSignature, ValueError):
+        except Exception:
+            # BadSignature, ValueError, or unpickling exceptions. If any of
+            # these happen, reset the session.
```
Comment (charettes) [inline django/contrib/sessions/backends/signed_cookies.py]: I couldn't find a valid way of making the `Exception` catching cloak possible misconfiguration but I wonder if it wouldn't be better to make session serializers re-raise a common `DeserializationError` and catch it here.

Comment (timgraham) [inline django/contrib/sessions/backends/signed_cookies.py]: My justification is that we are catching `Exception` in similar places of other cache backends. If it ends up hiding actual problems, maybe a solution could be to add debug logging.

Comment (charettes) [inline django/contrib/sessions/backends/signed_cookies.py]: I think debug logging would be fine here.

Comment (timgraham) [inline django/contrib/sessions/backends/signed_cookies.py]: Are you okay to defer it until someone reports an actual problem? Also, I question the usefulness of adding more logging until we change Django's default logging config so logged statements are actually displayed somewhere.

Comment (charettes) [inline django/contrib/sessions/backends/signed_cookies.py]: Ship it!

**Pull request merged**